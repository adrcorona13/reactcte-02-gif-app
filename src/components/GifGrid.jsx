import { GifGridItem } from "./GifGridItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GifGrid = ({category}) => {

    const {images, isLoading} = useFetchGifs(category);
    console.log({images, isLoading});


    
    return (
        <>
            <h3>{category}</h3>
            <div className="card-grid">
            {
                images.map((image ) => <GifGridItem key={image.id} 
                {...image} />)
            }
            </div>
        </>
    )
}