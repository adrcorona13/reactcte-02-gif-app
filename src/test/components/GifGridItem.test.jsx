import { render, screen } from "@testing-library/react"
import { GifGridItem } from "../../components/GifGridItem"

describe('Pruebas en GifGridItem', () => {

    const title = 'titulo';
    const url = 'https://url.com/';

    test('debe hacer match con el snapshot', () => {
        const { container } = render(<GifGridItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('debe mostrar la imagen con el url y el alt indicado', () => {
        render(<GifGridItem title={title} url={url} />);
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);
    });

    test('debe mostrar el titulo en el componente', () => { 
        render(<GifGridItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();

     })
})