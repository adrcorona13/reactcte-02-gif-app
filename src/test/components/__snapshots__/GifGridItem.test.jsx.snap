// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="titulo"
      src="https://url.com/"
    />
    <p>
      titulo
    </p>
  </div>
</div>
`;
